yield_plugin/src/yield_plugin.cpp

```diff
@@ -386,65 +400,118 @@ namespace yield_plugin
                                                                                                 initial_time,
                                                                                                 planning_time);
 
-    std::vector<double> original_traj_downtracks = get_relative_downtracks(original_tp);
+    std::vector<double> original_traj_relative_downtracks = get_relative_downtracks(original_tp);
     std::vector<double> calculated_speeds = {};
+    std::vector<double> new_relative_downtracks = {};
+    new_relative_downtracks.push_back(0.0);
     calculated_speeds.push_back(initial_velocity);
-    for(size_t i = 1; i < original_tp.trajectory_points.size(); i++ )
+    const auto original_planning_time = std::max(planning_time, (get_trajectory_end_time(original_tp) - get_trajectory_start_time(original_tp)));
+
+    // Calculating time_step to generate trajectory, which can be any constant.
+    // Only using original trajectory size here for potential performance improvement
+    const double average_time_step = std::max(0.1, original_planning_time / original_tp.trajectory_points.size() / 2);
```

Comment:
So the trajectory points are generated 1 meter apart, not by constant time. So the duration between points in original trajectory can vary - it can be higher than average_time_step between some points and lower for some. So the idea is that it is just arbitrarily dividing by 2 to capture more points that it may miss due to having slightly larger average_time_step. Or perhaps, the logic can iterate through and save the smallest duration between points in the original trajectory, which will capture all.